### PR TITLE
<Libc> Add support for scanf to accept formatted

### PR DESCRIPTION
**Pull request by sagarladla**
input

modified:   src/lib/libc/include/stdio.h
modified:   src/lib/libc/scanf.c

- Support for int, char, string, float format specifiers added for scanf
- Added gets function to accept string with spaces

Issue: #225

### CODE REVIEW

File: src/lib/libc/scanf.c [hunk 1]
```diff
@@ -42,3 +45,156 @@ char getchar()
...
+/**
+ * scanf
+ *
+ * @brief This function will take standard input for default data types.
+ *	  input will break at spaces
+ *
+ */
+
+#define CURRCHAR (temp)
+#define CHARIN() (temp = getch())
```
Comment (akashkollipara) [inline src/lib/libc/scanf.c:84]: Please avoid such macros.

File: src/lib/libc/scanf.c [hunk 2]
```diff
@@ -42,3 +45,156 @@ char getchar()
...
+
+/**
+ * get string with spaces
+ *
+ * @brief This function will take string input with spaces
+ *
+ */
+char *gets()
+{
+	char *s = (char *)malloc(sizeof(char)), temp;
```
Comment (akashkollipara) [inline src/lib/libc/scanf.c:57]: declare it in new line

Comment (akashkollipara) [inline src/lib/libc/scanf.c:57]: buffer overflow!!!!
allocating 1byte and then using it as array.

**Review by akashkollipara — CHANGES_REQUESTED** [review]

File: src/lib/libc/scanf.c [hunk 3]
```diff
@@ -183,8 +182,9 @@ int scanf(const char * restrict fmt, ...)
 				}
 				case 'f':
 				{
-					char *a = gets();
-					*(float *)va_arg(ap, float *) = (float)atof(a);
+					char buff[50];
```
Comment (akashkollipara) [inline src/lib/libc/scanf.c:185]: This is causes stack to bloat.
May result in stack corruption in small embedded systems.

Instead base on input args, like %d, %c, %s, etc, you may allocate a buffer.

%c need 1 byte. %d if its avr, d is for int which is 65535 max -> 5 bytes.
you will need such implementation and use malloc for the same.

**Review by akashkollipara — CHANGES_REQUESTED** [review]

File: src/lib/libc/scanf.c (hunk 1 above)
Comment (sagarladla) [reply src/lib/libc/scanf.c]: removed usage of macros

File: src/lib/libc/scanf.c (hunk 3 above)
Comment (sagarladla) [reply src/lib/libc/scanf.c]: Understood the general case, I need to re-structure the input buffer logic using malloc for each supported architecture. 

Then I can also check for out of range data type values.

**Pull request merged**